apps/web/vibes/soul/primitives/skeleton/index.tsx

```diff
@@ -0,0 +1,151 @@
+import { clsx } from 'clsx';
+
+/**
+ * This component supports various CSS variables for theming. Here's a comprehensive list, along
+ * with their default values:
+ *
+ * ```css
+ * :root {
+ *   --skeleton: hsl(var(--contrast-100));
+ * }
+ * ```
+ */
+
+function SkeletonBox({
+  height,
+  width,
+  rounded = 'none',
+}: {
+  height: string;
+  width?: string;
+  rounded?: 'none' | 'sm' | 'base' | 'md' | 'lg' | 'xl' | '2xl' | '3xl' | 'full';
+}) {
+  return (
+    <div
+      className={clsx(
+        'max-w-full bg-[var(--skeleton,hsl(var(--contrast-100)))]',
+        {
+          none: 'rounded-none',
+          sm: 'rounded-sm',
+          base: 'rounded',
+          md: 'rounded-md',
+          lg: 'rounded-lg',
+          xl: 'rounded-xl',
+          '2xl': 'rounded-2xl',
+          '3xl': 'rounded-3xl',
+          full: 'rounded-full',
+        }[rounded],
+      )}
+      style={{ height, width }}
+    />
+  );
+}
+
+SkeletonBox.displayName = 'Skeleton.Box';
+
+function SkeletonImage({
+  aspectRatio = '5:6',
+  height,
+  width,
+  rounded = 'none',
+}: {
+  aspectRatio?: '5:6' | '3:4' | '1:1';
+  height?: string;
+  width?: string;
+  rounded?: 'none' | 'sm' | 'base' | 'md' | 'lg' | 'xl' | '2xl' | '3xl' | 'full';
+}) {
+  return (
+    <div
+      className={clsx(
+        'max-w-full bg-[var(--skeleton,hsl(var(--contrast-100)))]',
+        {
+          '5:6': 'aspect-[5/6]',
+          '3:4': 'aspect-[3/4]',
+          '1:1': 'aspect-square',
+        }[aspectRatio],
+        {
+          none: 'rounded-none',
+          sm: 'rounded-sm',
+          base: 'rounded',
+          md: 'rounded-md',
+          lg: 'rounded-lg',
+          xl: 'rounded-xl',
+          '2xl': 'rounded-2xl',
+          '3xl': 'rounded-3xl',
+          full: 'rounded-full',
+        }[rounded],
+      )}
+      style={{ height, width }}
+    />
+  );
+}
+
+SkeletonImage.displayName = 'Skeleton.Image';
+
+function SkeletonText({
+  width = '10ch',
+  lineHeight = '1lh',
+  characterHeight = '1ex',
+  rounded = 'none',
+}: {
+  className?: string;
+  width?: string;
+  lineHeight?: string;
+  characterHeight?: string;
+  rounded?: 'none' | 'sm' | 'base' | 'md' | 'lg' | 'xl' | '2xl' | '3xl' | 'full';
+}) {
+  return (
+    <div className={clsx('flex items-center')} style={{ height: lineHeight }}>
+      <div
+        className={clsx(
+          'max-w-full bg-[var(--skeleton,hsl(var(--contrast-100)))]',
+          {
+            none: 'rounded-none',
+            sm: 'rounded-sm',
+            base: 'rounded',
+            md: 'rounded-md',
+            lg: 'rounded-lg',
+            xl: 'rounded-xl',
+            '2xl': 'rounded-2xl',
+            '3xl': 'rounded-3xl',
+            full: 'rounded-full',
+          }[rounded],
+        )}
+        style={{ width, height: characterHeight }}
+      />
+    </div>
+  );
+}
+
+SkeletonText.displayName = 'Skeleton.Text';
+
+function SkeletonPending({
```

Comment:
I'm still not following how this component is meant to be used in situations like `<ProductsListSection>` where the pending state isn't coupled to a skeleton. Or maybe it's not intended to be used there at all?